src/api/app/components/package_icon_and_link_component.html.haml

```diff
@@ -0,0 +1,3 @@
+= link_to package_show_path(@package.project, @package), class: 'text-word-break-all' do
```

Comment:
![Screenshot from 2022-02-17 14-43-17](https://user-images.githubusercontent.com/514785/154494470-8bafc068-e9cb-4b7c-b631-66e3c9dc3418.png)

Reply:
If there isn't any logic in this view component to be tested, then I agree with you @hennevogel. It is the case for those *IconAndLinkComponent.

Reply:
We created this to remove a big `if/else` in the parent component code. Let me see if we can remove those 3 components and not revert to that `if/else`...

Reply:
please don't ignore this part: "I think don't think this is needed in the UI"

Reply:
We are mixing two things here:

a) The header/title (the screenshot attached by @hennevogel before).
b) Each item of the list, which is currently generated by `*IconAndLinkComponent`.

Please, look at the red texts and arrows in this screenshot:

![watchlist](https://user-images.githubusercontent.com/2581944/154509314-ecc0c81b-5bbc-4b61-b901-4bf56d602c0f.png)

Having that clear...
I agree we can remove a) if we agree that it's enough to have the package name in the breadcrumb and URL. My screenshot is an example of a page where the package name only appears in those places.

Regarding b), we moved the code to view components for two main reasons: removing a big `if/else`, as Dani said, and also because we thought they can be reused in other parts of the application (the breadcrumbs for example, at least the project part for now).